go/api/base_client.go

```diff
@@ -69,14 +79,189 @@ func failureCallback(channelPtr unsafe.Pointer, cErrorMessage *C.char, cErrorTyp
 	resultChannel <- payload{value: nil, error: errors.GoError(uint32(cErrorType), msg)}
 }
 
+// RegisterClient registers a client in the registry using its pointer value
+func RegisterClient(client *baseClient, ptrValue uintptr) {
+	clientRegistryMu.Lock()
+	defer clientRegistryMu.Unlock()
+	clientRegistry[ptrValue] = client
+}
+
+// UnregisterClient removes a client from the registry
+func UnregisterClient(ptrValue uintptr) {
+	clientRegistryMu.Lock()
+	defer clientRegistryMu.Unlock()
+	delete(clientRegistry, ptrValue)
+}
+
+// GetClientByPtr gets a client from the registry by its pointer value
+func GetClientByPtr(ptrValue uintptr) *baseClient {
+	clientRegistryMu.RLock()
+	defer clientRegistryMu.RUnlock()
+	return clientRegistry[ptrValue]
+}
+
+//export pubSubCallback
+func pubSubCallback(clientPtr unsafe.Pointer, kind C.uint32_t, cResponse *C.struct_CommandResponse) {
+	defer C.free_command_response(cResponse)
```

Comment:
I think we have to keep callback fast - just copy data/pointer and return, in order to avoid delaying rust's threads
For refernce see callback on lines 68-80 and https://github.com/valkey-io/valkey-glide/blob/432ee6551196372026798996a78a415bb3a1dd9d/csharp/sources/Valkey.Glide/BaseClient.cs#L187-L190

Something like:
```
func pubSubCallback(...) {
  go func() {
    defer free(data)
    // handle data
  }()
}
```
Or use a channel for messages. Message handler will stay on the receiving end of that channel.

Reply:
Merged your suggested changes